lib/events.js

```diff
@@ -101,18 +103,34 @@ function emitThree(handler, isFn, self, arg1, arg2, arg3) {
   }
 }
 
+function emitMany(handler, isFn, self, args) {
+  if (isFn)
+    handler.apply(self, args);
+  else {
+    var len = handler.length;
+    var listeners = arrayClone(handler, len);
+    for (var i = 0; i < len; ++i)
+      listeners[i].apply(self, args);
+  }
+}
+
 EventEmitter.prototype.emit = function emit(type) {
-  var er, handler, len, args, i, listeners, events, domain;
+  var er, handler, len, args, i, events, domain, doError;
   var needDomainExit = false;
 
   events = this._events;
-  if (!events)
-    events = this._events = {};
+  if (!events) {
+    if (type === 'error')
+      doError = true;
+    else
+      return false;
+  } else
+    doError = (type === 'error' && !events.error);
```

Comment:
This can be made a little flatter and more readable by swapping the consequent and the alternate.  Suggestion:

```javascript
if (events)
  doError = (type === 'error' && events.error != null);
else if (type === 'error')
  doError = true;
else
  return false;
```

You can condense it but I'll leave it up to you to decide what's more readable.

```javascript
doError = (type === 'error');
if (events)
  doError = (doError && events.error == null);
else if (!doError)
  return false;
```